src/test/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperationTest.java

```diff
@@ -929,5 +941,176 @@ public void failureUnorderedAll() throws Exception {
+                  Values.of(CustomValueSerializers.getIntegerMapValues(shredDocument.arraySize())),
+                  Values.of(CustomValueSerializers.getStringMapValues(shredDocument.arrayEquals())),
+                  Values.of(
+                      CustomValueSerializers.getStringSetValue(shredDocument.arrayContains())),
+                  Values.of(
+                      CustomValueSerializers.getBooleanMapValues(shredDocument.queryBoolValues())),
+                  Values.of(
+                      CustomValueSerializers.getDoubleMapValues(shredDocument.queryNumberValues())),
+                  Values.of(
+                      CustomValueSerializers.getStringMapValues(shredDocument.queryTextValues())),
+                  Values.of(CustomValueSerializers.getSetValue(shredDocument.queryNullValues())),
+                  Values.of(
+                      CustomValueSerializers.getTimestampMapValues(
+                          shredDocument.queryTimestampValues())),
+                  CustomValueSerializers.getVectorValue(shredDocument.queryVectorValues()))
+              .withColumnSpec(
+                  List.of(
+                      QueryOuterClass.ColumnSpec.newBuilder()
+                          .setName("applied")
+                          .setType(TypeSpecs.BOOLEAN)
+                          .build()))
+              .withSerialConsistency(queriesConfig.serialConsistency())
+              .returning(List.of(List.of(Values.of(true))));
+
+      InsertOperation operation = new InsertOperation(COMMAND_CONTEXT_VECTOR, shredDocument);
+      Supplier<CommandResult> execute =
+          operation
+              .execute(queryExecutor)
+              .subscribe()
+              .withSubscriber(UniAssertSubscriber.create())
+              .awaitItem()
+              .getItem();
+
+      // assert query execution
+      insertAssert.assertExecuteCount().isOne();
+
+      // then result
+      CommandResult result = execute.get();
+      assertThat(result.status())
+          .hasSize(1)
+          .containsEntry(CommandStatus.INSERTED_IDS, List.of(new DocumentId.StringId("doc1")));
+      assertThat(result.errors()).isNull();
+    }
+
+    @Test
+    public void insertOneVectorDisabledWithVectorData() throws Exception {
+      String document =
+          """
+        {
+          "_id": "doc1",
+          "text": "user1",
+          "number" : 10,
+          "boolean": true,
+          "nullval" : null,
+          "array" : ["a", "b"],
+          "sub_doc" : {"col": "val"},
+          "date_val" : {"$date": 1672531200000 },
+          "$vector" : [0.11,0.22,0.33,0.44]
+        }
+        """;
+
+      JsonNode jsonNode = objectMapper.readTree(document);
+      WritableShreddedDocument shredDocument = shredder.shred(jsonNode);
+      InsertOperation operation = new InsertOperation(COMMAND_CONTEXT, shredDocument);
+      Throwable failure = catchThrowable(() -> operation.execute(queryExecutor));
+      assertThat(failure)
```

Comment:
Does this mean you can't insert a document with a `$vector` property unless vector search is enabled for the collection you're trying to insert to?

Reply:
Yes, $vector field cannot be inserted unless collection created with vector search index enabled.